### PR TITLE
#65: re-factor controllers and respective tests to use async postpone…

### PR DESCRIPTION
**Pull request by andrii-kovalenko-ct**
- All the controllers (except Health) are refactored to return CompletionStage
  - respective tests are re-factored to use Spring async MVC mocking
    - added special `MockMvcAsync` class to wrap common async operations
  - re-factored most of the test to use `executeBlocking()` instead of `.toCompletableFuture().executeBlocking()`

### CODE REVIEW

**Comment by lojzatran** [comment]
Fixes: 
https://github.com/commercetools/commercetools-paypal-plus-integration/issues/35
https://github.com/commercetools/commercetools-paypal-plus-integration/issues/45

File: src/main/java/com/commercetools/payment/handler/CommercetoolsHealthController.java [hunk 1]
```diff
@@ -30,11 +33,23 @@ public CommercetoolsHealthController(@Nonnull StringTrimmerEditor stringTrimmerE
             method = RequestMethod.GET,
             value = "/health",
             produces = APPLICATION_JSON_VALUE)
-    public Map<String, Object> checkHealth() {
-        Map<String, Object> tenantResponse = new HashMap<>();
-        tenantResponse.put("tenants", this.tenantProperties.getTenants().keySet());
-        tenantResponse.put("statusCode", HttpStatus.OK.value());
-        return tenantResponse;
+    public CompletionStage<Map<String, Object>> checkHealth() {
```
[the remark this answers is not in the document]
Comment (andrii-kovalenko-ct) [reply src/main/java/com/commercetools/payment/handler/CommercetoolsHealthController.java]: it was a wrong commit, fixed 79aaab4

File: src/main/java/com/commercetools/payment/handler/CommercetoolsHealthController.java [hunk 2]
```diff
@@ -30,11 +33,23 @@ public CommercetoolsHealthController(@Nonnull StringTrimmerEditor stringTrimmerE
...
+
+        CompletableFuture<Map<String, Object>> mapCompletableFuture = CompletableFuture.supplyAsync(() -> {
+
+            try {
+                TimeUnit.SECONDS.sleep(5);
```
[the remark this answers is not in the document]
Comment (andrii-kovalenko-ct) [reply src/main/java/com/commercetools/payment/handler/CommercetoolsHealthController.java]: it was a wrong commit, fixed 79aaab4

File: src/main/java/com/commercetools/payment/handler/CommercetoolsPatchPaymentsController.java [hunk 3]
```diff
@@ -28,13 +30,14 @@ public CommercetoolsPatchPaymentsController(@Nonnull StringTrimmerEditor stringT
             method = RequestMethod.POST,
             value = "/{tenantName}/commercetools/patch/payments/{ctpPaymentId}",
             produces = APPLICATION_JSON_VALUE)
-    public ResponseEntity patchPayment(@PathVariable String tenantName,
-                                       @PathVariable String ctpPaymentId) {
-        PaymentHandleResponse paymentHandleResponse = paymentHandlerProvider
+    public CompletionStage<ResponseEntity> patchPayment(@PathVariable String tenantName,
+                                                        @PathVariable String ctpPaymentId) {
+        CompletionStage<PaymentHandleResponse> paymentHandleResponse = paymentHandlerProvider
```
[the remark this answers is not in the document]
Comment (andrii-kovalenko-ct) [reply src/main/java/com/commercetools/payment/handler/CommercetoolsPatchPaymentsController.java]: good point!
b3b96c4

**Pull request merged**